### PR TITLE
fix(Table): Fix autoscroll not showing given row

### LINKED ISSUE #1003: Table: auto scroll (scrollToRow) doesn't scroll to the needed row
#### Describe the bug (actual behavior)
Auto scroll is not working properly. 
When your needed row is below the viewport, auto-scroll scrolls to the row. And that row is either cut (just a half or less of it is visible) or it is 1 row below (not visible at all).
When the needed row is above the viewport,  auto scroll works perfectly.

#### Expected behavior
Auto-scrolled row should be fully visible.

The best would be if there was an option to choose to have the auto-scrolled row at the beginning of the viewport. Now it is either the beginning or the end of the viewport.
Same feature implemented in `itwin/ui-components` using `react-window`. Has a good api on how to scroll:
"auto" | "smart" | "center" | "end" | "start" 
https://github.com/iTwin/itwinjs-core/blob/912f966315168c861a33da4aaf41e86668c5cf11/ui/components/src/ui-components/tree/controlled/component/TreeRenderer.tsx#L86

#### Reproduction
`iTwinUI-react: 1.48.1`: https://codesandbox.io/s/itwinui-react-example-forked-6ud3uo?file=/src/App.tsx
`iTwinUI-react: 2.2.1`: https://codesandbox.io/s/itwinui-react-example-forked-0stjvo?file=/src/App.tsx

##### Steps to reproduce
1. Scroll to the top of the table
2. In `input` field enter number (e.,g. 150) you want to scroll to and press `scroll to row` button
3. The last visible row is 149 instead of 150

#### Additional information
`react: 16.14.0`

`iTwinUI-react: 1.48.1`
image
`iTwinUI-react: 2.2.1`
image

### PR DESCRIPTION
**Pull request by elephantcatdog**
## Changes
Added code to update the visible index after scrolling to the index. Then the next two functions that check if the row is fully visible will work properly. Previously they were just running with the old start and end of `visibleIndex`.

## Testing
- Ran all tests
- I looked into adding another test in `VirtualScroll.test.tsx`, but I don't think it's possible. You can't update properties in tests unless you rerender, but rerendering means that the `scrollToIndex` value is always initiated vs being updated like in this issue. When `VirtualScroll` is initialized with `scrollToIndex`, the index is initialized to be in the center of the space vs the end or beginning.
- I instead just tested it with the repro code given in the issue (https://codesandbox.io/s/itwinui-react-example-forked-0stjvo?file=/src/App.tsx), put into `App.tsx` (vite). Then tested looking for a row off screen down, off screen up, and on screen.

## Docs
- Added changeset

closes #1003

### CODE REVIEW

**Comment by LostABike** [comment]
Do you have the link to this bug? Or how do I reproduce it?

**Comment by elephantcatdog** [comment]
@LostABike The related issues are linked in the side nav of github: 
image

**Comment by elephantcatdog** [comment]
@LostABike I included a link in the description of the repro(duction) that the user sent in with the issue (codesandbox.io/s/itwinui-react-example-forked-0stjvo?file=/src/App.tsx](https://codesandbox.io/s/itwinui-react-example-forked-0stjvo?file=/src/App.tsx).

To reproduce it, you can put the code in this repro in the file `playgrounds/vite/src/App.tsx` in your branch.

**Review by LostABike — COMMENTED** [review]
I have read and manually tested this change and everything works ok. But maybe wait for another approval just to sanity check as I'm fairly new ;D

**Comment by elephantcatdog** [comment]
@LostABike Will do. Don't be afraid to approve PRs if they look good to you. We wait for two approvals to merge, and the reviewers are split into two groups - more experienced and less experienced. So there will always be someone else that will review.

**Comment by elephantcatdog** [comment]
> Do you have the link to this bug? Or how do I reproduce it?

@LostABike  I forgot to add the 'closes #xxx', sorry

**Review by gretanausedaite — APPROVED** [review]
Looks good.
Table has scroll flicker but I believe there is a separate issue for that.

**Comment by gretanausedaite** [comment]
@LostABike You should click approve if you think that PR is good to be merged. We encourage people to wait for two approvals before merging their PRs.

File: .changeset/spotty-kiwis-brake.md
Comment (elephantcatdog) [reply .changeset/spotty-kiwis-brake.md:2]: Would that say `'@itwin/itwinui-react': patch` or is the line removed for a patch?

Comment (gretanausedaite) [reply .changeset/spotty-kiwis-brake.md]: Yes, change minor to patch.

**Comment by GintV** [comment]
@elephantcatdog Since it is a bug fix, can this be backported to 1.x ? Currently we are forced to use 1.x because of iTwin.js Core 3.x.

**Comment by mayank99** [comment]
@GintV We have decided against backporting currently. We are working with the App UI team to consume iTwinUI 2.x in their 4.0 version, so that would be the path forward for you.

In the meantime, you are able to use [`patch-package`](https://www.npmjs.com/package/patch-package) or [`pnpm patch`](https://pnpm.io/cli/patch) to edit your node modules.

**Comment by GintV** [comment]
@mayank99 We will try to use itwinui 2.x together with iTwin.js 3.x, but if that does not work out, it won't be acceptable for us to wait for iTwin.js 4.x since it will be at least 3 months before it is released + stabilization period for being early birds

**Pull request merged**
